manifests/init.pp

```diff
@@ -34,127 +32,65 @@
-  #### v5 only ####
-  $use_generic_template = $filebeat::params::use_generic_template,
-  $shutdown_timeout     = $filebeat::params::shutdown_timeout,
-  $beat_name            = $filebeat::params::beat_name,
-  $tags                 = $filebeat::params::tags,
-  $queue_size           = $filebeat::params::queue_size,
-  $max_procs            = $filebeat::params::max_procs,
-  $fields               = $filebeat::params::fields,
-  $fields_under_root    = $filebeat::params::fields_under_root,
-  $processors           = {},
-  $processors_merge     = false,
-  #### End v5 only ####
-  $prospectors          = {},
-  $prospectors_merge    = false,
-  $proxy_address        = undef
+  String  $package_ensure       = $filebeat::params::package_ensure,
+  Boolean $manage_repo          = $filebeat::params::manage_repo,
+  Variant[Boolean, Enum['stopped', 'running']] $service_ensure = $filebeat::params::service_ensure,
+  Boolean $service_enable = $filebeat::params::service_enable,
+  Optional[String]  $service_provider = $filebeat::params::service_provider,
+  Optional[Integer] $repo_priority = undef,
+  Integer $spool_size           = $filebeat::params::spool_size,
+  String  $idle_timeout         = $filebeat::params::idle_timeout,
+  Boolean $publish_async        = $filebeat::params::publish_async,
+  String  $registry_file        = $filebeat::params::registry_file,
+  String  $config_file          = $filebeat::params::config_file,
+  String[4,4]  $config_dir_mode = $filebeat::params::config_dir_mode,
+  String  $config_dir           = $filebeat::params::config_dir,
+  String[4,4]  $config_file_mode = $filebeat::params::config_file_mode,
+  Boolean $purge_conf_dir       = $filebeat::params::purge_conf_dir,
+  Hash    $outputs              = $filebeat::params::outputs,
+  Hash    $shipper              = $filebeat::params::shipper,
+  Hash    $logging              = $filebeat::params::logging,
+  Hash    $run_options          = $filebeat::params::run_options,
+  Optional[String]  $conf_template        = $filebeat::params::conf_template,
+  Optional[Pattern[/^(http(?:s)?\:\/\/[a-zA-Z0-9]+(?:(?:\.|\-)[a-zA-Z0-9]+)+(?:\:\d+)?(?:\/[\w\-]+)*(?:\/?|\/\w+\.[a-zA-Z]{2,4}(?:\?[\w]+\=[\w\-]+)?)?(?:\&[\w]+\=[\w\-]+)*)$/]] $download_url         = undef,
+  Optional[String]  $install_dir          = $filebeat::params::install_dir,
+  String  $tmp_dir              = $filebeat::params::tmp_dir,
+  Integer $shutdown_timeout     = $filebeat::params::shutdown_timeout,
+  String  $beat_name            = $filebeat::params::beat_name,
+  Array   $tags                 = $filebeat::params::tags,
+  Integer $queue_size           = $filebeat::params::queue_size,
+  Optional[Integer] $max_procs            = $filebeat::params::max_procs,
+  Hash $fields               = $filebeat::params::fields,
+  Boolean $fields_under_root    = $filebeat::params::fields_under_root,
+  Boolean $disable_config_test  = $filebeat::params::disable_config_test,
+  Hash    $processors           = {},
```

Comment:
Thanks for catching this - I was looking at what I put in the docs and the default value.